### PR TITLE
Image: add meilisearch v1.2.0

### PR DESCRIPTION
**Pull request by jamonation**
### Pre-review Checklist

- [ ] **IMPORTANT: 'image-request' tag has been applied if this PR is adding any images, including new versions or variants**

#### Quality Requirements for Images PRs
The items in this checklist should all be checked in the PR with exceptions clearly documented.
The general idea is that to the extent possible, the image should be a drop-in replacement to its public counterpart.
- [X] Image is a smaller size than its common public counterpart, or if not, the reason why should be explained
- [X] Image should be scanned for CVEs and should return 0 or near-0 results. Exceptions should be explained.
- [X] The software in the image should be able to be installed and work as expected
- [X] build for x86_64 and aarch64 if possible, document exceptions

#### For new image PRs only

If you have an apko.yaml file in this PR you need to follow this checklist, otherwise feel free to remove.
- [X] Include tests, sufficient enough that you would trust this image running in production.
- [X] Perform manual tests as needed and document your testing results
  - [X] For services images (rabbit, databases, webservers) test basic functionality, upstream install/getting started, port availability, admin access. Document differences from public image.

- [X] The version included is the latest GA version of the software
- [X] The latest tag points to the newest stable version
- [X] The image runs as `nonroot` and GID/UID are set to 65532 or upstream default
  - [X] Alternatively the username and GID/UID may be a commonly used one from the ecosystem e.g: postgres
- [X] ENTRYPOINT
  - [X] For applications/servers/utilities call main program with no arguments e.g. [redis-server]
- [X] CMD:
  - [X] For server applications give arguments to start in daemon mode (may be empty)
  - [X] For utilities/tooling bring up help e.g. `–help`
- [X] Add annotations

- [X] Ensure the image responds to SIGTERM
  - [X] `docker kill $(docker run -d --rm cgr.dev/chainguard/meilisearch)`
- [X] Error logs write to stderr and normal logs to stdout. DO NOT write to file.
#### Documentation Requirements

- [X] Provide a README file, follow the [README template](readme-template.md) to include variants, tags, and usage examples
- [X] Document where and how the image deviates from popular alternatives (ports, packages, user, entrypoint, versions, etc). Explain why there's a difference and if the user needs to do anything differently when using the image

### CODE REVIEW

File: images/meilisearch/tests/02-index.sh [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+	fi
+	EXIT=0
+}
+
+function docker_stop() {
+	docker stop -t 0 "$ID"
+	exit $EXIT
+}
+
+preflight
```
Comment (imjasonh) [inline images/meilisearch/tests/02-index.sh:44]:
```suggestion
```

**Review by imjasonh — COMMENTED** [review]
Looking good! Just a few comments about the tests, which we haven't standardized or documented very well. Otherwise, this is great. Thanks @jamonation !

File: images/meilisearch/tests/02-index.sh [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+EXIT=1
+
+function preflight() {
+	if [[ "${IMAGE_NAME}" == "" ]]; then
+		echo "Must set IMAGE_NAME environment variable. Exiting."
+		exit 1
+	fi
+}
+
```
[the remark this answers is not in the document]
Comment (jamonation) [reply images/meilisearch/tests/02-index.sh:13]: Ah nice, no more need for me to cargo cult this 😁

File: images/meilisearch/tests/02-index.sh [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+function docker_stop() {
+	docker stop -t 0 "$ID"
+	exit $EXIT
+}
+
+preflight
+run_meilisearch
+test
+docker_stop
```
[the remark this answers is not in the document]
Comment (jamonation) [reply images/meilisearch/tests/02-index.sh:47]: Will do

**Comment by jamonation** [comment]
Think I got this all tidied @imjasonh, thanks for the quick walk through on how to use TF for testing too

**Review by imjasonh — APPROVED** [review]

**Pull request merged**